### PR TITLE
Include 'email bounced' codes in unredeemed filter

### PR DESCRIPTION
**Pull request by mushtaqak**
This includes codes with `OFFER_ASSIGNMENT_EMAIL_BOUNCED` error status to appear in Unredeemed filter toggle. 

[ENT-1607](https://openedx.atlassian.net/browse/ENT-1607)

### CODE REVIEW

File: ecommerce/extensions/api/v2/tests/views/test_enterprise.py [hunk 1]
```diff
@@ -784,6 +784,50 @@ def test_coupon_codes_detail_with_pagination(self, page, page_size, pagination,
...
+
+        # Verify that code appears in unredeemed filter.
+        self.assert_code_detail_response(
+            response['results'],
+            [{'code': 0, 'assigned_to': 'user1@example.com', 'redemptions': {'used': 0, 'total': 1}}],
+            codes
+        )
+
+        # Email bounce a code.
+        assignment = OfferAssignment.objects.filter(code=vouchers[0].code).first()
```
Comment (saleem-latif) [inline ecommerce/extensions/api/v2/tests/views/test_enterprise.py]: You can simplify the update here

```
OfferAssignment.objects.filter(code=vouchers[0].code).update(status=OFFER_ASSIGNMENT_EMAIL_BOUNCED)
```

**Review by saleem-latif — APPROVED** [review]
One comment, overall LGTM 👍

**Review by muhammad-ammar — APPROVED** [review]
LGTM 👍

**Comment by mushtaqak** [comment]
@brittneyexline Can you review and merge if it's all good ?

**Review by brittneyexline — APPROVED** [review]

**Pull request merged**